unyt/array.py

```diff
@@ -166,6 +168,15 @@ def _iterable(obj):
     return True
 
 
+def _unit_operation_error_raise_or_warn(ufunc, u0, u1, func, *inputs):
+    if u0.registry.strict and u1.registry.strict:  # True by default
+        raise UnitOperationError(ufunc, u0, u1)
+    else:
+        warnings.warn(UnitOperationWarning(ufunc, u0, u1))
+        unwrapped_inputs = [i.value if isinstance(i, unyt_array) else i for i in inputs]
+        return func(*unwrapped_inputs)
```

Comment:
So I can see two problems here
- `unyt_array` doesn't have a `registry` attribute. I think you mean `u0.units.registry`
- even applying the naive fix I get a failing test (in a case where u1 is a pure np.ndarray)

I think I got the fix by reversing the conditional and adding a more complete condition (all tests pass on my machine with this patch)
```suggestion
    if (
        hasattr(u0, "units")
        and not u0.units.registry.strict
        and hasattr(u1, "units")
        and not u1.units.registry.strict
    ):
        warnings.warn(UnitOperationWarning(ufunc, u0, u1))
        unwrapped_inputs = [i.value if isinstance(i, unyt_array) else i for i in inputs]
        return func(*unwrapped_inputs)
    else:
        raise UnitOperationError(ufunc, u0, u1)
```
I have further comments I want to make here but for now let's focus on passing tests, then we'll refactor :)

Reply:
Thank you kindly for all the comments! These should be resolved now, happy to discuss refactor/changes